### PR TITLE
switch-dav1d: 1.5.0

### PR DESCRIPTION
**Pull request by dragonflylee**

### CODE REVIEW

File: switch/dav1d/dav1d.patch [hunk 1]
```diff
@@ -15,35 +15,35 @@ index 1104aef..17b804b 100644
...
 --- a/src/cpu.c
 +++ b/src/cpu.c
-@@ -36,6 +36,9 @@
- #elif defined(__APPLE__)
+@@ -37,6 +37,9 @@
+ #ifdef __APPLE__
  #include <sys/sysctl.h>
  #include <sys/types.h>
 +#elif defined(__SWITCH__)
 +#undef PACKED // libdav1d's definition is incompatible with libnx' usage of the macro
```
Comment (averne) [inline switch/dav1d/dav1d.patch:26]: This can be dropped, recent versions of libnx have prefixed the macro with NX_ which solves the issue.

**Comment by averne** [comment]
@dragonflylee Were you able to test this patch? I built the library and tested a couple files with my player, which reliably either fail decoding, or trigger a segfault. These files work fine with dav1d 1.2.1 from the devkitpro repos. I attach the files if you want to confirm with your application.
[av1_film_grain.webm](https://github.com/user-attachments/assets/96dc08ab-53f6-43a6-a2e5-7ee3adc05c8d) (fails to parse frame headers)
[av1_srgb_still_picture.ivf.zip](https://github.com/user-attachments/files/17480219/av1_srgb_still_picture.ivf.zip) (segfaults, decompress the zip)

Also, the checksum on the source tarball was outdated, which leads me to think the release is not quite stable yet.

**Comment by dragonflylee** [comment]
sorry for mismatch of checksum, here is my [PKGBUILD](https://github.com/dragonflylee/switchfin/blob/dev/scripts/switch/dav1d/PKGBUILD)

tested [av1_film_grain.webm](https://github.com/user-attachments/assets/96dc08ab-53f6-43a6-a2e5-7ee3adc05c8d) with https://nightly.link/dragonflylee/switchfin/actions/runs/11461808437/Switchfin-deko3d-dd829a4.zip and play ok

**Comment by averne** [comment]
> sorry for mismatch of checksum, here is my [PKGBUILD](https://github.com/dragonflylee/switchfin/blob/dev/scripts/switch/dav1d/PKGBUILD)
> 
> tested [av1_film_grain.webm](https://github.com/user-attachments/assets/96dc08ab-53f6-43a6-a2e5-7ee3adc05c8d) with https://nightly.link/dragonflylee/switchfin/actions/runs/11461808437/Switchfin-deko3d-dd829a4.zip and play ok

Thanks for the confirmation. Going through the changelog I realized dav1d broke its ABI, after recompiling ffmpeg I can confirm libdav1d 1.5.0 currently in the dkp repos decodes these samples fine. Sorry for the false report.

**Comment by WinterMute** [comment]
Would you mind PRing mpv, ffmpeg & harfbuzz when you get a chance?

**Comment by averne** [comment]
I have patches for an mpv audio backend using audren which I can PR here. I also have a deko3d graphics backend, but it relies on a runtime port of uam (https://github.com/averne/libuam), so that would need to get added to the dkP repos as well, but I don't know if it is acceptable to you (iiuc that's not the vision fincs had for runtime shader compilation).

As for ffmpeg, I'll PR an updated version of the patch I've continued to maintain. Fyi, I also tried to upstream it, but it understandably got rejected for containing to much low-level details. I'm working an a library to separate os-level stuff from decode-specific parts. That would also allow me to support desktop nvidia cards which would enable automated testing for popular codecs.

**Comment by averne** [comment]
That's awesome to hear, thanks for the work

**Pull request merged**